scripts/check-exports.mjs

```diff
@@ -0,0 +1,151 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* eslint-env node */
+
+import fs from 'fs';
+import { parseArgs } from 'util';
+import { createRequire } from 'module';
+import chalk from 'chalk';
+import upath from 'upath';
+import { parse } from 'acorn';
+import { walk } from 'estree-walker';
+import { globSync } from 'glob';
+import constants from './release/utils/constants.js';
+
+const require = createRequire( import.meta.url );
+
+const {
+	CKEDITOR5_ROOT_PATH,
+	CKEDITOR5_COMMERCIAL_PATH,
+	PACKAGES_DIRECTORY
+} = constants;
+
+/**
+ * List of paths to the allowed `input` packages.
+ */
+const paths = {
+	'ckeditor5': upath.join( CKEDITOR5_ROOT_PATH, 'src', 'index.ts' ),
+	'ckeditor5-premium-features': upath.join( CKEDITOR5_COMMERCIAL_PATH, 'packages', 'ckeditor5-premium-features', 'src', 'index.ts' )
+};
```

Comment:
Could we extend https://github.com/ckeditor/ckeditor5/blob/master/scripts/release/utils/constants.js instead of creating a new, custom map?